src/object/interrupt.c

```diff
@@ -182,21 +182,22 @@ void deletedIRQHandler(irq_t irq)
 void handleInterrupt(irq_t irq)
 {
     if (unlikely(IRQT_TO_IRQ(irq) > maxIRQ)) {
-        /* mask, ack and pretend it didn't happen. We assume that because
-         * the interrupt controller for the platform returned this IRQ that
-         * it is safe to use in mask and ack operations, even though it is
-         * above the claimed maxIRQ. i.e. we're assuming maxIRQ is wrong */
-        printf("Received IRQ %d, which is above the platforms maxIRQ of %d\n", (int)IRQT_TO_IRQ(irq), (int)maxIRQ);
+        /* The interrupt number is out of range. Pretend it did not happen by
+         * handling it like an inactive interrupt (mask and ack). We assume this
+         * is acceptable, because the platform specific interrupt controller
+         * driver reported this interrupt. Maybe the value maxIRQ is just wrong
+         * or set to a lower value because the interrupts are unused.
+         */
+        printf("Received IRQ %d, which is above the platforms maxIRQ of %d\n",
+               (int)IRQT_TO_IRQ(irq), (int)maxIRQ);
         maskInterrupt(true, irq);
         ackInterrupt(irq);
         return;
     }
+
     switch (intStateIRQTable[IRQT_TO_IDX(irq)]) {
     case IRQSignal: {
-        cap_t cap;
-
-        cap = intStateIRQNode[IRQT_TO_IDX(irq)].cap;
-
+        cap_t cap = intStateIRQNode[IRQT_TO_IDX(irq)].cap;
```

Comment:
The proofs did unfortunately fail on this one. Probably the reason it was written this way in the first place.

Reply:
I don't think it was the reason - the kernel used to be compiled with C89 not C99 which lead to this style. 

I recall asking about it with some patches and while inlining the declaration and definition in new code is fine, changing all of the existing sites where this is done was going to be a huge amount of effort.

Reply:
Added a comment about this and created https://github.com/seL4/seL4/issues/576 as a reminder to look into this one day